src/MountedTraversal.js

```diff
@@ -177,6 +178,12 @@ export function parentsOfInst(inst, root) {
   return pathToNode(inst, root).reverse();
 }
 
+export function instMatchesObjectProps(inst, props) {
+  if (!isDOMComponent(inst)) return false;
+  const node = getNode(inst);
+  return !isEmpty(props) && isMatch(propsOfNode(node), props);
```

Comment:
Actually, i'm leaning more towards throwing an error for an empty object.

I don't think it should be encouraged to use a selector with an empty objects. There are better ways to find nodes and a more descriptive selector should be used. 

If I looked at tests and saw `wrapper.find({})`. It'd be quite confused on what it was trying to find.